Tasks/VsTest/vstest.ts

```diff
@@ -100,30 +101,18 @@ catch (error) {
     throw error;
 }
 
-function getResolvedPattern(pattern: string): string {
-    if (path.isAbsolute(pattern)) {
-        return pattern;
-    }
-    else {
-        return path.join(systemDefaultWorkingDirectory, pattern);
+function getTestAssemblies(): string[] {
+    if (isNullOrWhitespace(searchFolder)) {
+        searchFolder = systemDefaultWorkingDirectory;
+        tl.debug("Search directory empty, defaulting to " + searchFolder);
     }
-}
-
-function getPatternWithoutIncludeExclude(pattern: string): string {
-    return pattern.startsWith('-:') || pattern.startsWith('+:') ? pattern.substr(2) : pattern;
-}
-
-function getFilteredFilesFromPattern(pattern: string, hasQuantifier: boolean, allFiles: string[]): string[] {
-    return hasQuantifier ? getFilteredFiles(pattern, allFiles) : [pattern];
-}
-
-function getTestAssemblies(): Set<string> {
-    return new Set(ffl.findFiles(testAssembly, false, systemDefaultWorkingDirectory));
+    tl.debug("Searching for test assemblies in: " + searchFolder);
+    return tl.findMatch(searchFolder, testAssembly, null, getTaskMatchOption());
```

Comment:
nit: the verbiage in the UI seems to indicate `searchFolder` is the backstop - e.g. calling `find()` on `searchFolder` and then separately calling `match()`.

however, by calling `findMatch`, `searchFolder` will effectively be the "default root" for any unrooted patterns. so you could potentially end up with files outside `searchFolder`. iirc, this is the behavior that you actually want. so my only feedback here is you may consider tweaking the vebiage in the UI.